src/pages/index.js

```diff
@@ -4,54 +4,148 @@ import Section from "../components/Section.js";
 
 const pictureModal = new PopupWithImage(".modal-picture");
 pictureModal.setEventListeners();
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      cardSection.addItem(createCard(cardData));
-    },
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
+// const cardSection = new Section(
+//   {
+//     items: initialCards,
+//     renderer: (cardData) => {
+//       cardSection.addItem(createCard(cardData));
+//     },
+//   },
+//   ".cards__list"
+// );
+// cardSection.renderItems();
+
+const deleteModal = new PopupWithConfirm(".modal-delete");
+deleteModal.setEventListeners();
 
 const userInfo = new UserInfo({
   userNameSelector: ".profile__title",
-  userJobSelector: ".profile__description",
+  userAboutSelector: ".profile__description",
+  userAvatarSelector: ".profile__image",
 });
+
+let userId;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then((values) => {
+    userId = values[0]._id;
+    const userData = values[0];
+
+    const initialCards = values[1];
+
+    userInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+      avatar: userData.avatar,
+    });
+    const cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: (cardData) => {
+          cardSection.addItem(createCard(cardData));
+        },
+      },
+      ".cards__list"
+    );
+    cardSection.renderItems();
+    // return userData;
+  })
+  .catch((err) => {
+    console.error("Error. The request has failed: ", err);
+  });
+
 const profileModal = new PopupWithForm(".modal-profile", (inputs) => {
-  userInfo.setUserInfo(inputs);
-  profileModal.close();
+  profileModal.changeButtonText("Saving...");
+  api
+    .updateUserInfo(inputs)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        about: res.about,
+        avatar: res.avatar,
+      });
+      modalFormProfile.reset();
+      profileModal.close();
+      profileModal.changeButtonText("Save");
+    })
+    .catch((err) => {
+      console.error("Error. The request has failed: ", err);
+      profileModal.changeButtonText("Save");
+    });
 });
 profileModal.setEventListeners();
 
+const avatarModal = new PopupWithForm(".modal-avatar", (inputs) => {
+  avatarModal.changeButtonText("Saving...");
+  console.log(inputs.avatar);
+
+  api
+    .updateProfilePicture(inputs.avatar)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        about: res.about,
+        avatar: res.avatar,
+      });
+      modalFormAvatar.reset();
+      avatarModal.close();
+      avatarModal.changeButtonText("Save");
+    })
+    .catch((err) => {
+      console.error("Error. The request has failed: ", err);
+      avatarModal.changeButtonText("Save");
+    });
+});
+avatarModal.setEventListeners();
+const modalFormAvatar = document.querySelector(".modal__form-avatar");
+const avatarFormValidator = new FormValidator(config, modalFormAvatar);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```